.github/workflows/release.yml

```diff
@@ -24,3 +24,30 @@ jobs:
           args: release --clean
         env:
           GITHUB_TOKEN: ${{secrets.GH_PAT}}
+  
+  docker:
+    runs-on: ubuntu-latest
+    needs: goreleaser
+    if: ${{ secrets.DOCKER_HUB_USERNAME && secrets.DOCKER_HUB_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
+
+      - name: Build and Push Docker image
+        uses: docker/build-push-action@v6
+        platforms: linux/amd64,linux/arm64
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: |
+            go-task/task:latest
```

Comment:
Since this workflow is triggered on tag creation, it would be a good idea to tag the created image with the version tag that was created in git. That way we can have matching tags for the docker images and the git repo.

We can still leave in a `latest` tag, but I consider actual version tags to be a requirement to properly use this image